app/controllers/advertisements_controller.rb

```diff
@@ -25,7 +25,7 @@ def show
       @advertisement_html = render_advertisement
       @campaign_url = advertisement_clicks_url(@virtual_impression_id, campaign_id: @campaign.id)
       @impression_url = impression_url(@virtual_impression_id, template: template_name, theme: theme_name, format: :gif)
-      @powered_by_url = root_url(ref: @virtual_impression_id)
+      @powered_by_url = invite_url(User.referral_code(@campaign.user_id))
```

Comment:
Should this use the publisher ID and not the campaign user ID?

Reply:
👍 Great catch. Thanks.